ntbs-service/wwwroot/source/Components/ValidateTravelOrVisit.ts

```diff
@@ -0,0 +1,135 @@
+                            // Here it's convenient to act on 'Property' only
+                            if (key.indexOf(".") === -1) {
+                                const baseRef = key.charAt(0).toLowerCase() + key.substring(1);
+                                const errorRef = baseRef + "ErrorRef";
+                                const formGroupRef = baseRef + "FormGroup";
+                                const errorMessage = data[key];
+
+                                const baseElement = this.$refs[baseRef];
+                                if (baseElement.tagName === "INPUT") {
+                                    this.$refs[baseRef].classList.add("nhsuk-input--error");
+                                } else if (baseElement.tagName === "SELECT") {
+                                    this.$refs[baseRef].classList.add("nhsuk-select--error");
+                                }
+
+                                this.$refs[errorRef].textContent = errorMessage;
+                                this.$refs[formGroupRef].classList.add("nhsuk-form-group--error");
+                            }
+                        }
+                    }
+                });
+        },
+
+        resetErrors: function () {
+            const refList = [
+                "totalNumberOfCountries",
+                "country1Id",
+                "country2Id",
+                "country3Id",
+                "stayLengthInMonths1",
+                "stayLengthInMonths2",
+                "stayLengthInMonths3"
+            ];
+
+            for (let i = 0; i < refList.length; i++) {
+                const ref = refList[i];
+                const errorRef = ref + "ErrorRef";
+                const formGroupRef = ref + "FormGroup";
+
+                const baseElement = this.$refs[ref];
+                if (baseElement.tagName === "INPUT") {
+                    this.$refs[ref].classList.remove("nhsuk-input--error");
+                } else if (baseElement.tagName === "SELECT") {
+                    this.$refs[ref].classList.remove("nhsuk-select--error");
+                }
+
+                this.$refs[errorRef].textContent = "";
+                this.$refs[formGroupRef].classList.remove("nhsuk-form-group--error");
+            }
+        },
+
+        getTravelOrVisitVariables: function () {
+            const variables: TravelOrVisitVariables = ({
+                hasData: this.getHasDataValue(),
+                totalNumberOfCountries: this.$refs["totalNumberOfCountries"].value,
+                country1Id: this.$refs["country1Id"].value,
+                country2Id: this.$refs["country2Id"].value,
+                country3Id: this.$refs["country3Id"].value,
+                stayLengthInMonths1: this.$refs["stayLengthInMonths1"].value,
+                stayLengthInMonths2: this.$refs["stayLengthInMonths2"].value,
+                stayLengthInMonths3: this.$refs["stayLengthInMonths3"].value
+            }) as {
+                hasData: string;
```

Comment:
Why do we need the `as` cast?
It suggests we're not typing something properly

Reply:
Fixed and tested still working